recipes/bitserializer-rapidjson/all/conanfile.py

```diff
@@ -0,0 +1,39 @@
+from conans import ConanFile, tools
+import os
+
+
+class BitserializerConan(ConanFile):
+    name = "bitserializer-rapidjson"
+    description = "C++ 17 library for serialization, module for support JSON (implementation based on the RapidJson library)"
+    topics = ("serialization", "json")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://bitbucket.org/Pavel_Kisliak/bitserializer"
+    license = "MIT"
+    settings = ("compiler",)
+    no_copy_source = True
+    requires = ("bitserializer/0.9", "rapidjson/[>=1.1.0]")
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def configure(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, "17")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        # Find and rename folder in the extracted sources
+        # This workaround used in connection that zip-archive from BitBucket contains folder with some hash in name
+        for dirname in os.listdir(self.source_folder):
+            if "-bitserializer-" in dirname:
+                os.rename(dirname, self._source_subfolder)
+                break
+
+    def package(self):
+        include_folder = os.path.join(self._source_subfolder, "archives")
+        self.copy(pattern="license.txt", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="bitserializer_rapidjson\\*.h", dst="include", src=include_folder)
+
+    def package_id(self):
+        self.info.header_only()
```

Comment:
you may want to set the cmake find package variabled to match [your cmake](https://bitbucket.org/Pavel_Kisliak/bitserializer/src/00cb2ea0bdc725f6c0ac4be563e5a4deb2e7666d/archives/bitserializer_rapidjson/CMakeLists.txt#lines-21) there will be a future release with components support

Reply:
@prince-chrismc Are you suggesting to add method such next:
```
    def package_info(self):
        self.cpp_info.names["cmake_find_package"] = "BitSerializer::bitserializer-rapidjson"
        self.cpp_info.names["cmake_find_package_multi"] = "BitSerializer::bitserializer-rapidjson"
```
Currently I'm not use my own cmake scripts for build Conan package, not sure am I right, this is my first experience with Conan. But as currently my library is header-only and does not have some hard build logic it seems not bad.

Reply:
Sadly I don't think that is possible currently, or at least I've never seen it. 

I misread the make on my first pass, I thought the component was `BitSerializer-RapidJson`  but since it's all lower case you will be okay with the default